README.md

```diff
@@ -40,16 +40,93 @@ The Rucio file catalog is much flatter than that of a usual `POSIX` filesystems
 - Fix Mac OS X mouting issues
 - Surely a lot more...
 
-## How to build
+## Getting Started
+
+### Cloning the repository
+Fork the repository and clone it on your local machine for development
+```BASH
+$ git clone https://github.com/<your-username>/fuse-posix.git
+$ cd fuse-posix/
+$ git remote add upstream https://github.com/rucio/fuse-posix.git
+```
+### Install the dependencies
+Next step is to install the dependencies which shall be required to build the software
+
+**Ubuntu/Debian (apt)**
+```BASH
+$ sudo apt-get install cmake git g++ libfuse-dev libcurl4-gnutls-dev
+```
+**CentOS (yum)**
+```BASH
+$ yum install -y cmake3 libcurl-devel libfuse-devel
+```
+
+### Build the software
 To build the software please run:
 
 ```[shell]
-./build.sh
+$ cd fuse-posix/
+$ ./build.sh
+```
+**NOTE:** To complete the build `libcurl-devel`, `fuse-libs` and `fuse-devel` packages (or equivalent) must be present:
+`cmake` will try to locate them for you and trigger some build messages if unable to do so.
+Please note that `cmake` version 3 or greater is needed.
+
+### Setting up the FUSE mount
+If you're Ubuntu 18.04 LTS or above, you should particularly follow this step to ensure the right working of the Rucio-FUSE mount.
+* **STEP 1** - Check FUSE version
+```BASH
+$ fusermount -V
+fusermount version: 2.9.7
+```
+Also check if the `fuse` group is available on the system and your current user is a part of it or not
+```BASH
+$ groups $USER | grep fuse
+```
+If this highlights `fuse` then your current USER is a part of the `fuse` group and you may skip STEP 2. If not, then add the `fuse` group as explained in STEP 2.
+
+* **STEP 2** - Add `fuse` group for current USER
+```BASH
+$ sudo groupadd fuse
+$ sudo usermod -aG fuse $USER
+```
+The above step adds a new group named `fuse` and makes the current user a part of it. If you wish to access the FUSE mount as `root`, then you must also uncomment the line `user_allow_other` in `/etc/fuse.conf` to enable root access for FUSE filesystem. Once this is done successfully, reload the fuse module with `modprobe` or restart the machine.
+```
+$ modprobe fuse
 ```
 
+### Mounting with Rucio-FUSE mount
+Now that `fusermount` is set up on the machine, we can proceed with the mounting process. The first step is to edit the `settings.json.template` and add the authentication details.
+
+```BASH
+$ mv settings.json.template settings.json
+$ gedit settings.json #or use vim/vi
```

Comment:
A bit specific: all the line of code in the doc must be executable, meaning that you cannot comment. In addition `mv` is not great since you loose the template. I'd just rephrase into something in the lines of "Create a settings.json file based on the template found at settings.json.template" and purge the bash section.

Reply:
Yes, it will be helpful to preserve the template

Reply:
Changed it accordingly